plugins/bigquery/dbt/adapters/bigquery/impl.py

```diff
@@ -143,9 +143,27 @@ def truncate_relation(self, relation: BigQueryRelation) -> None:
     def rename_relation(
         self, from_relation: BigQueryRelation, to_relation: BigQueryRelation
     ) -> None:
-        raise dbt.exceptions.NotImplementedException(
-            '`rename_relation` is not implemented for this adapter!'
-        )
+        self.cache_renamed(from_relation, to_relation)
```

Comment:
Can you move this down to just before the copy/delete calls? I think we want to avoid updating the cache until we're just about to do it, if possible.